presidio-anonymizer/presidio_anonymizer/anonymizers/mask.py

```diff
@@ -1,19 +1,59 @@
 """Mask some or all given text entity PII with given character."""
 from presidio_anonymizer.anonymizers import Anonymizer
+from presidio_anonymizer.anonymizers.validators import validate_parameter
+from presidio_anonymizer.entities import InvalidParamException
 
 
-# TODO implement + test
 class Mask(Anonymizer):
     """Mask the given text with given value."""
 
-    def anonymize(self, original_text=None, params={}):
+    def anonymize(self, text: str = None, params: dict = None) -> str:
         """
-        Anonymize a given amount of text with a given character.
+        Mask a given amount of text with a given character.
 
-        :return: The given text masked as requested
+        :param text: the text to be masked
+        :param params:
+            masking_char: The character to be masked with
+            chars_to_mask: The amount of characters to mask
+            from_end: Whether to mask the text from it's end
+        :return: the masked text
         """
-        # chars_to_replace = params.get("chars_to_replace")
-        # from_end = params.get("from_end")
-        # replace_with = params.get("replace_with")
-        # old_text = params.get("old_text")
-        return ""
+        effective_chars_to_mask = self._get_effective_chars_to_mask(
+            text, params.get("chars_to_mask")
+        )
+        from_end = params.get("from_end")
+        masking_char = params.get("masking_char")
+        return self._get_anonymized_text(
+            text, effective_chars_to_mask, from_end, masking_char
+        )
+
+    def validate(self, params: dict = None) -> None:
+        """
+        Validate the parameters for mask.
+
+        :param params:
+            masking_char: The character to be masked with
+            chars_to_mask: The amount of characters to mask
+            from_end: Whether to mask the text from it's end
+        """
+        masking_char = params.get("masking_char")
+        validate_parameter(masking_char, "masking_char", str)
+        if len(masking_char) > 1:
+            raise InvalidParamException(
+                f"Invalid input, masking_char must be a character"
```

Comment:
you can extract the word "masking_char" into a static variable in the class.
It will prevent mistakes of spelling it over and over again.
I think that's what @navalev  suggested with the consts